ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/CompactorMR.java

```diff
@@ -237,6 +237,7 @@ void run(HiveConf conf, String jobName, Table t, Partition p, StorageDescriptor
     }
 
     JobConf job = createBaseJobConf(conf, jobName, t, sd, writeIds, ci);
+    QueryCompactor.Util.removeAbortedDirsForAcidTable(conf, dir);
```

Comment:
@vpnvishv Why do we do this here? I understand we can, but why don't we let the Cleaner to delete the files? This just makes the compactor slower. Do we have a functionality reason for this?
After this change it will run in CompactorMR and in MMQueryCompactors, but not in normal QueryCompactors?

Reply:
@pvargacl You are right, this is not required, as now compactor run in a transaction and the cleaner has validTxnList with aborted bits set. This we have added wrt to Hive-3, in which cleaner doesn't have aborted bits set, as we create validWriteIdList for cleaner based on the highestWriteId.

Reply:
removed